Source/JavaScriptCore/wasm/WasmTypeDefinition.h

```diff
@@ -216,8 +216,108 @@ enum Mutability : uint8_t {
     Immutable = 0
 };
 
-struct FieldType {
-    Type type;
+struct StorageType {
+public:
+    template <typename T>
+    bool is() const { return std::holds_alternative<T>(m_storageType); }
+
+    template <typename T>
+    const T as() const { ASSERT(is<T>()); return *std::get_if<T>(&m_storageType); }
```

Comment:
I adjusted `as()` to return `T` instead of `T*` and changed all callsites as it was too easy to make mistakes (e.g., checking equality on the pointers and not the actual values) and it introduced a lot of noise of `*` dereferences.